dev/releases/update_website.py

```diff
@@ -138,124 +139,112 @@ def mb_bytes(nrbytes):
-# Use the GitHub API to find the GAP release on GitHub
-# TODO do this via the python GitHub module
-github_releases_url = 'https://api.github.com/repos/' + args.gap_fork + '/gap/releases'
-request = requests.get(github_releases_url)
-try:
-    request.raise_for_status()
-except:
-    error('failed to connect to GitHub API at ' + github_releases_url)
-
-# Select the appropriate release (as a dictionary)
-releases = request.json()
-releases = [ x for x in releases if is_possible_gap4_release_tag(x['tag_name']) ]
-if len(releases) == 0:
-    error("no GAP v4.X.Y releases found on GitHub!")
-if args.tag == None:
-    # Using latest release; therefore ignore drafts and prerelease
-    release = [ x for x in releases if not x['draft'] and not x['prerelease']]
-    if not release:
-        error("cannot determine the latest (non-draft, non-prerelease) release")
-    release = release[0]
-else:
-    release = [ x for x in releases if x['tag_name'] == args.tag ]
-    if len(release) != 1:
+# Use the GitHub API to find the appropriate GAP release on GitHub
+utils.CURRENT_REPO_NAME = os.environ.get("GITHUB_REPOSITORY", args.gap_fork + "/gap")
+utils.initialize_github(args.token)
+if args.tag:
+    try:
+        release = utils.CURRENT_REPO.get_release(args.tag)
+    except:
         error("non-existent or ambiguous release tag name: " + args.tag)
-    release = release[0]
+else:
+    # Using latest release; therefore ignore drafts and prerelease
+    release = [ x for x in utils.CURRENT_REPO.get_releases() if \
+                not x.draft and not x.prerelease and is_possible_gap4_release_tag(x.tag_name)]
+    try:
+        release = release[0]
+    except:
+        error("cannot determine the latest (non-draft, non-prerelease) v4.X.Y release")
 
 # Divide the assets betwen Windows and UNIX assets
-# TODO should archives e.g.  packages-required-v4.11.0.tar.gz be in assets_unix?
-assets = release['assets']
+# TODO should archives e.g. packages-required-v4.11.0.tar.gz be in assets_unix?
+# TODO Ultimately the assets data should be added to a YAML/JSON data file in
+# the GapWWW repo, and parsed/displayed by the website setup
+assets = release.get_assets()
 def is_windows_asset(name):
-    return any(name.endswith(x) for x in ['exe', '-win.zip', '-win32.zip'])
+    return any(name.endswith(x) for x in [".exe", "-win.zip", "-win32.zip"])
 assets_windows = []
 assets_unix = []
 for asset in assets:
-    if not asset['name'].endswith('sha256'):
-        if is_windows_asset(asset['name']):
+    if not asset.name.endswith("sha256"):
+        if is_windows_asset(asset.name):
             assets_windows.append(asset)
         else:
             assets_unix.append(asset)
 
 # Release date
 if args.use_github_date:
-    release_date = dateutil.parser.isoparse(release['published_at'])
-year  = release_date.strftime('%Y')
-month = release_date.strftime('%B')
-date  = release_date.strftime('%d %B %Y')
+    release_date = release.published_at
```

Comment:
What type has `release_date` at this point? Is it a string? A python date object? (I have an ulterior motive for asking, but I think perhaps it would be good to record this as a comment here for the future anyhow)

Reply:
Dates that you get via PyGitHub seem to always be Python `datetime` objects. In particular that is the case here.

Reply:
I have added such a comment.